ndscan/experiment/fragment.py

```diff
@@ -317,59 +317,77 @@ def setattr_param(self, name: str, param_class: Type, description: str, *args,
             ``<original_owner>.<original_name>``). If ``None``, defaults to ``name``.
+        :param additional_owners: List of additional owners of the parameter to rebind.
+            Each element in the list should be a tuple consisting of a fragment and
+            optionally a parameter name (defaults to ``original_name``` if missing or
+            `None`). Metadata from these parameters will not be inherited by the new
+            parameter.
         :param kwargs: Any attributes to override in the parameter metadata, which
             defaults to that of the original parameter.
         :return: The newly created parameter handle.
         """
+        original_name = original_name if original_name is not None else name
         assert (self._building
                 ), "Can only call setattr_param_rebind() during build_fragment()"
         assert name.isidentifier(), "Parameter name must be valid Python identifier"
         assert not hasattr(self, name), "Field '{}' already exists".format(name)
-
-        if original_name is None:
-            original_name = name
         assert hasattr(original_owner, original_name), \
             "Original owner does not have a field of name '{}'".format(original_name)
-        assert original_name in original_owner._free_params, (
-            "Field '{}' is not a free parameter of original owner; "
-            "already rebound?".format(original_name))
+        assert name not in self._rebound_subfragment_params
 
-        # Set up our own copy of the parameter.
         original_param = original_owner._free_params[original_name]
-        param = deepcopy(original_param)
-        param.fqn = self.fqn + "." + name
+        new_param = deepcopy(original_param)
+        new_param.fqn = self.fqn + "." + name
         for k, v in kwargs.items():
-            setattr(param, k, v)
-        self._free_params[name] = param
-        handle = param.HandleType(self, name)
-        setattr(self, name, handle)
-
-        # Deregister it from the original owner and make sure we set the store
-        # to our own later.
-        original_handles = original_owner._get_all_handles_for_param(original_name)
-        del original_owner._free_params[original_name]
-        assert name not in self._rebound_subfragment_params
-        self._rebound_subfragment_params[name] = original_handles
-
-        return handle
+            setattr(new_param, k, v)
+        self._free_params[name] = new_param
+        new_handle = new_param.HandleType(self, name)
+        setattr(self, name, new_handle)
+
+        owners = [(original_owner, original_name)]
+        if additional_owners is not None:
+            owners += additional_owners
+
+        self._rebound_subfragment_params[name] = []
+        for owner, *owner_name in owners:
+            [owner_name] = owner_name if owner_name != [] else [original_name]
+            owner_name = owner_name if owner_name is not None else original_name
```

Comment:
@lochsh is there an obviously cleaner way of expressing the logic around this?

Reply:
What about 
```python
owner_name = original_name if not owner_name or not owner_name[0] else owner_name
```